src/hooks/paginated/useLoadSitePolygonData.ts

```diff
@@ -1,123 +1,83 @@
+  // Validate entity type
+  const entityName = useMemo(() => {
+    if (!entityType || !VALID_ENTITY_TYPES.includes(entityType as any)) {
+      throw new Error(`Invalid entityType: ${entityType}. Must be one of: ${VALID_ENTITY_TYPES.join(", ")}`);
     }
+    return entityType as "sites" | "projects";
+  }, [entityType]);
 
-    abortControllerRef.current = new AbortController();
-    const signal = abortControllerRef.current.signal;
-
-    try {
-      setLoading(true);
-
-      if (signal.aborted) return;
-
-      const { count } = await fetchGetV2EntityPolygonsCount({
-        queryParams: {
-          uuid: entity_uuid,
-          type: entity_type,
-          status: statuses ?? "",
-          [`sort[${sortOrder}]`]: sortOrder === "created_at" ? "desc" : "asc",
-          ...(validFilter !== "all" && { valid: validFilter })
-        }
-      });
-      if (signal.aborted) return;
+  // Build filter object for the new API
+  const filter = useMemo(() => {
+    const filterObj: Record<string, unknown> = {};
 
-      setTotal(count!);
-      let result: any[] = [];
-      const limit = 500;
-      let offset = 0;
+    if (statuses) {
+      const statusArray = statuses.split(",").filter(Boolean);
+      if (statusArray.length > 0) {
+        filterObj.polygonStatus = statusArray;
+      }
+    }
 
-      while (offset < count! && !signal.aborted) {
-        const queryParams: any = {
-          limit: limit,
-          offset: offset,
-          uuid: entity_uuid,
-          type: entity_type,
-          status: statuses ?? "",
-          lightResource: true,
-          [`sort[${sortOrder}]`]: sortOrder === "created_at" ? "desc" : "asc",
-          ...(validFilter !== "all" && { valid: validFilter })
-        };
+    if (validFilter && validFilter !== VALID_FILTER_ALL) {
+      filterObj.validationStatus = [validFilter];
+    }
 
-        const partialResponse = (await fetchGetV2EntityPolygons({
-          queryParams
-        })) as any;
+    return filterObj;
+  }, [statuses, validFilter]);
 
-        if (signal.aborted) return;
+  const {
+    data: sitePolygonsData,
+    isLoading,
+    total,
+    refetch,
+    updateSinglePolygon
+  } = useAllSitePolygons({
+    entityName,
+    entityUuid,
+    enabled: !!entityUuid,
+    filter
+  });
 
-        result = result.concat(partialResponse);
-        setData(result as any);
+  const [polygonCriteriaMap, setPolygonCriteriaMap] = useState<Record<string, unknown>>({});
 
-        setProgress(Math.min(offset + limit, count!));
-        offset += limit;
-      }
+  const updateSingleSitePolygonData = (old_id: string, updatedData: SitePolygonFullDto) => {
+    try {
+      updateSinglePolygon(old_id, updatedData);
     } catch (error) {
-      if (error instanceof Error && error.name !== "AbortError") {
-        console.error("Error in loadInBatches:", error);
-      }
-    } finally {
-      if (!signal.aborted) {
-        setLoading(false);
-      }
+      refetch();
```

Comment:
This seems like it could use a call to `Log.error` so the problem bubbles up to sentry.